src/controllers/userInfo.controller.ts

```diff
@@ -23,6 +24,40 @@ class UserInfoController {
 		});
 		return res.send(updatedInfo);
 	}
+
+	async updateAlpacaToken(req: Request, res: Response) {
+		const clientId = process.env.CLIENT_ID as string
+		const clientSecret = process.env.CLIENT_SECRET as string
+		const redirectURI = process.env.REDIRECT_URI as string
+		let code = req.body.code;
+		const userId = new mongoose.Types.ObjectId(req.params.userId);
+		const params = new URLSearchParams();
+		params.append("grant_type", "authorization_code");
+		params.append("code", code);
+		params.append("client_id", clientId);
+		params.append("client_secret", clientSecret);
+		params.append("redirect_uri", redirectURI);
+
+		try {
+			const response = await axios.post("https://api.alpaca.markets/oauth/token", params, {
+				headers: {
+					"Content-Type": "application/x-www-form-urlencoded",
+				},
+			});
+			const alpacaToken = response.data.access_token;
```

Comment:
1. If you only need a specific data from an axios response, you can get it directly from the curly bracket as below. 
2. We should have a condition to check, though it is super rare, the access-token is not null/undefined.

```suggestion
			const {data.access_token} = await axios.post("https://api.alpaca.markets/oauth/token", params, {
				headers: {
					"Content-Type": "application/x-www-form-urlencoded",
				},
			});
			if(!access_token){
			   return res.status(500).send("No token");
			}
			
			const alpacaToken = access_token;
```

Reply:
good point